test/test_target.py

```diff
@@ -663,6 +663,66 @@ def test_glibc_bessel_functions(dtype):
                                rtol=1e-6, atol=1e-6)
 
 
+def test_c_vector_extensions():
+    knl = lp.make_kernel(
+        "{[i, j1, j2, j3]: 0<=i<10 and 0<=j1,j2,j3<4}",
+        """
+        <> temp1[j1] = x[i, j1]
+        <> temp2[j2] = 2*temp1[j2] + 1 {inames=i:j2}
+        y[i, j3] = temp2[j3]
+        """,
+        [lp.GlobalArg("x, y", shape=lp.auto, dtype=float)],
+        seq_dependencies=True,
+        target=lp.CVectorExtensionsTarget())
+
+    knl = lp.tag_inames(knl, "j2:vec, j1:ilp, j3:ilp")
+    knl = lp.tag_array_axes(knl, "temp1,temp2", "vec")
+
+    print(lp.generate_code_v2(knl).device_code())
+
+
+def test_omp_simd_tag():
+    knl = lp.make_kernel(
+        "{[i]: 0<=i<16}",
+        """
+        y[i] = 2 * x[i]
+        """)
+
+    knl = lp.add_dtypes(knl, {"x": "float64"})
+    knl = lp.split_iname(knl, "i", 4)
+    knl = lp.tag_inames(knl, {"i_inner": lp.OpenMPSIMDTag()})
+
+    code_str = lp.generate_code_v2(knl).device_code()
+
+    assert any(line.strip() == "#pragma omp simd"
+               for line in code_str.split("\n"))
+
+
+def test_vec_tag_with_omp_simd_fallback():
+    knl = lp.make_kernel(
+        "{[i, j1, j2, j3]: 0<=i<10 and 0<=j1,j2,j3<4}",
+        """
+        <> temp1[j1] = x[i, j1]
+        <> temp2[j2] = 2*temp1[j2] + 1 {inames=i:j2}
+        y[i, j3] = temp2[j3]
+        """,
+        [lp.GlobalArg("x, y", shape=lp.auto, dtype=float)],
+        seq_dependencies=True,
+        target=lp.CVectorExtensionsTarget(),
+        lang_version=(2018, 2))
+
+    knl = lp.tag_inames(knl, {"j1": lp.VectorizeTag(lp.OpenMPSIMDTag()),
+                              "j2": lp.VectorizeTag(lp.OpenMPSIMDTag()),
+                              "j3": lp.VectorizeTag(lp.OpenMPSIMDTag())})
```

Comment:
@inducer: Any big red signals with the user-facing interface of specifying fallbacks?

Reply:
I think `OpenMPSIMDTag` shoudl be renamed to something more generic, and then this fallback could be automatic.

Reply:
Yep, this has been reimplemented.